src/module/apps/party/sheet.js

```diff
@@ -45,6 +45,13 @@ class PTUPartySheet extends FormApplication {
         data.boxed = this.boxed;
         data.available = this.available;
 
+        const averageLevelOfMons = monArray => (monArray.map(p => p.system.level.current).reduce((a, b) => a + b) / monArray.length).toFixed(1);
+        const hasMon = monArray => Array.isArray(monArray) && monArray.length > 0
+
+        data.partyApl = hasMon(this.party) ? averageLevelOfMons(data.party) : undefined
+        data.boxedApl = hasMon(this.boxed)  ? averageLevelOfMons(data.boxed) : null
+        data.availableApl = hasMon(this.available)  ? averageLevelOfMons(data.available) : 0
```

Comment:
```suggestion
        const averageLevelOfMons = monArray => (monArray.reduce((a, b) => (a.attributes.level.current ?? 0) + (b.attributes.level.current ?? 0)) / monArray.length).toFixed(1);

        data.partyApl = this.party?.length > 0 ? averageLevelOfMons(data.party) : undefined
        data.boxedApl = this.boxed?.length > 0  ? averageLevelOfMons(data.boxed) : undefined
        data.availableApl = this.available?.length > 0  ? averageLevelOfMons(data.available) : undefined
```
- Removed the redundant map
- Added null check on level
  - Should only occur if data preperation fails, unlikely scenario but one that is achievable.
- Removed Array.isArray check as it is redundant, if length > 1 it should be itterable.
- Equalized all to be `undefined` instead of a random Falsey value

@Tinkastrong why did you assign them all a different falsey value? While it may be fine this time; JS handles `undefined`, `null` and `0` very differently from each other in many situations, so please for the future stick to one unless there is a very good reason not to, to circumvent future bugs.

Reply:
regarding the falsy values - I'm sorry for that  - I apparently rolled back a change from when checking if they behave differently.

How should a mon without `system.level?.current` be handled? 
1. Ignored and act as if it is not present in the array
2. Handled as if level 0
3. Make the APL calculation stop for that APL.

Reply:
As you can see in the above suggested changes I put it as `?? 0`.

Reply:
<- not used to GitHub UI. mb